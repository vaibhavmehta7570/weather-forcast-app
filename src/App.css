@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@100;200;300;400;500;600;700;800;900&display=swap');

body {
  margin: 0;
  padding: 0;
  color: #ffffff;
  background-color: #363546;
}

input:focus,
select:focus,
textarea:focus,
button:focus {
  outline: none;
}

main {
  text-align: center;
}

#root {
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-family: 'Poppins', sans-serif;
}

.navbar {
  overflow: hidden;
  background-color: #24252a;
  padding: 0 2rem;
  font-size: 1rem;
}

.navbar a {
  float: left;
  display: block;
  text-align: center;
  text-decoration: none;
  color: #ffffff;
  padding: 1rem 1rem;
}

.navbar a:hover {
  background-color: #333333;
}

.active {
  background-color: #469bfc;
}

.home__background {
  margin-top: 10rem;
  width: 30%;
  height: auto;
}
