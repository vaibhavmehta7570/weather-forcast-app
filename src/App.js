import React from 'react'
import { Route, Switch } from 'react-router-dom'
import Home from './Home'
import Counter from './task1/components/counter'
import WeatherApp from './task2/components/WeatherApp'
import Navbar from './navbar'

const App = () => {
  return (
    <React.Fragment>
      <Navbar />
      <main>
        <Switch>
          <Route exact path='/'>
            <Home />
          </Route>
          <Route path='/counter'>
            <Counter />
          </Route>
          <Route path='/weatherapp'>
            <WeatherApp />
          </Route>
        </Switch>
      </main>
    </React.Fragment>
  )
}

export default App
