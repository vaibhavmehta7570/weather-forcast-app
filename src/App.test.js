import React from 'react'
import App from './app'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

describe('should test app.js', () => {
  test('should render correctly app component', () => {
    const appRenderResponse = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>
      )
      .toJSON()
    expect(appRenderResponse).toMatchSnapshot()
  })
})
