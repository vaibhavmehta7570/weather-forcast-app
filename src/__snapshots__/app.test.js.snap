// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test app.js should render correctly app component 1`] = `
Array [
  <nav
    className="navbar"
  >
    <a
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/counter"
      onClick={[Function]}
    >
      Counter Task
    </a>
    <a
      href="/weatherapp"
      onClick={[Function]}
    >
      Weather App
    </a>
  </nav>,
  <main />,
]
`;
